pkg/controllers/hub/trafficmanagerbackend/controller.go

```diff
@@ -240,10 +244,33 @@ func (r *Reconciler) handleUpdate(ctx context.Context, backend *fleetnetv1alpha1
 	klog.V(2).InfoS("Found the serviceImport", "trafficManagerBackend", backendKObj, "serviceImport", klog.KObj(serviceImport), "clusters", serviceImport.Status.Clusters)
 
 	desiredEndpointsMaps, invalidServicesMaps, err := r.validateExportedServiceForServiceImport(ctx, backend, serviceImport)
-	if err != nil {
+	if err != nil || (desiredEndpointsMaps == nil && invalidServicesMaps == nil) {
+		// We don't need to requeue not found internalServiceExport(err == nil and desiredEndpointsMaps == nil && invalidServicesMaps == nil)
+		// as when the serviceImport is updated, the controller will be re-triggered again.
+		// The controller will retry when err is not nil.
 		return ctrl.Result{}, err
 	}
 	klog.V(2).InfoS("Found the exported services behind the serviceImport", "trafficManagerBackend", backendKObj, "serviceImport", klog.KObj(serviceImport), "numberOfDesiredEndpoints", len(desiredEndpointsMaps), "numberOfInvalidServices", len(invalidServicesMaps))
+
+	acceptedEndpoints, err := r.updateTrafficManagerEndpoints(ctx, backend, atmProfile, desiredEndpointsMaps)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	if len(invalidServicesMaps) > 0 {
+		for clusterID, invalidServiceErr := range invalidServicesMaps {
+			message := fmt.Sprintf("%v service(s) exported from clusters cannot be exposed as the Azure Traffic Manager, for example, service exported from %v is invalid: %v", len(invalidServicesMaps), clusterID, invalidServiceErr)
+			setFalseCondition(backend, acceptedEndpoints, message)
+			// Here we only populate the message with the first invalid exported service.
+			// Note, the loop of the invalidServicesMaps is not deterministic.
+			break
+		}
+	} else {
+		setTrueCondition(backend, acceptedEndpoints)
```

Comment:
In `updateTrafficManagerEndpoints`, there's chance that the endpoint is not updated due to clientErr, we set the condition to false and return nil. Then here because the err is nil, we set the condition to True.